pkg/compute/watchers/executor_watcher.go

```diff
@@ -40,11 +41,7 @@ func (h *ExecutionUpsertHandler) HandleEvent(ctx context.Context, event watcher.
 	case models.ExecutionStateNew:
 		if err := h.bidder.RunBidding(ctx, execution); err != nil {
 			compute.ExecutionBiddingErrors.Add(ctx, 1)
-			logger.Error().
-				Err(err).
-				Msg("failed to run bidding")
-			return fmt.Errorf("failed to handle execution state %s: %w",
-				execution.ComputeState.StateType, err)
+			return bacerrors.Wrap(err, "failed to handle execution state %s", execution.ComputeState.StateType)
```

Comment:
still having inconsistent error handling behaviour. Here you are returning bacerror. in `executor.Run`, you are logging and returning raw error. and in `executor.Cancel` you are returning bacerror. 

Lets just have a single error handling logic at the end such as
```
	if err != nil {
		return bacerrors.Wrap(err, "failed to handle execution state %s", execution.ComputeState.StateType)
	}
```